### PR TITLE
[README]Add the build instruction

### LINKED ISSUE #48: Build instructions doc

### PR DESCRIPTION
**Pull request by minggangw**
This patch added the instruction which will show the developers how to
setup the environment on different platforms, install the build tool
chain, and build this nodejs module itself.

Besides the build instruction, it also included some other instructions,
like running unit test, a simple code snippet, API specification and the
license we use.

Fix #48

### CODE REVIEW

**Comment by minggangw** [comment]
@qiuzhong @Honry @kenny-y would you help me to review this instruction to see if it has described the whole process properly, thanks!

File: README.md [hunk 1]
```diff
@@ -1,8 +1,78 @@
...
+
+Currently, the ROS is a cross-platform system, which covers Linux, MacOS and Windows, and the `rclnodejs` module is developed on the [`master`](https://github.com/ros2/ros2/blob/master/ros2.repos) branch of ROS, so you have to build ROS from source at the present stage. Please select the platform you want to work on, then reference the instruction of building ROS.
+
+* [Linux](https://github.com/ros2/ros2/wiki/Linux-Development-Setup)
+* [MacOS](https://github.com/ros2/ros2/wiki/OSX-Development-Setup)
+* [Windows](https://github.com/ros2/ros2/wiki/Windows-Development-Setup)
+
+2.Install `Node.js`.
+
+The `Node.js` version we selected is the latest LTS `Boron`. You can install it:
```
Comment (qiuzhong) [inline README.md]: Suggest to tell the user least number version(6.x here) of Node.js, which is more intuitive.

File: README.md [hunk 2]
```diff
@@ -1,8 +1,78 @@
...
+
+* Download from Node.js offical [website](https://nodejs.org/en/), and install it.
+* Use the Node Version Manager ([nvm](https://github.com/creationix/nvm)) to install it.
+
+3.Install `node-gyp`, the build tool for `rclnodejs`.
+
+```javascript
+npm install -g node-gyp
```
Comment (qiuzhong) [inline README.md]: Actually, from npm 2.15.12, `node-gyp` was brought to `Node.js` 4.x. See [this change](https://github.com/npm/npm/blob/771c66884a0a1851cffb3bf132828f6ccef1c09c/changelogs/CHANGELOG-2.md#v21512-2017-03-24).

File: README.md [hunk 3]
```diff
@@ -1,8 +1,78 @@
...
+```
+
+## Build Module
+
+This `Node.js` module is built by [node-gyp](https://www.npmjs.com/package/node-gyp), all you have to do is just to run the following command (**We only support the Linux platform in the current phase**):
+
+```javascript
```
Comment (qiuzhong) [inline README.md:29]: The documentation above is telling users to build ROS2 from source, and it's with `--symlink-install` parameter by default. In that case, users should run `source install/local_setup.bash` first.

**Review by qiuzhong — COMMENTED** [review]
@minggangw , check my reviews.

File: README.md [hunk 4]
```diff
@@ -1,8 +1,78 @@
...
+npm install
+```
+
+## Run Unit Test
+
+[mocha](https://www.npmjs.com/package/mocha) is a javascript test framework for node.js, simply run the following commond to run the unit test under `test` folder:
```
Comment (Honry) [inline README.md]: typo, "commond"

File: README.md (hunk 1 above)
Comment (minggangw) [reply README.md]: That' clear, thanks.

File: README.md (hunk 2 above)
Comment (minggangw) [reply README.md]: Wow, I've not noticed this, so `node-gyp` doesn't need be mentioned here.

File: README.md (hunk 3 above)
Comment (minggangw) [reply README.md:29]: That' right, thanks.

File: README.md (hunk 4 above)
Comment (minggangw) [reply README.md]: Oh, I will fix it, thanks.

**Pull request merged**